js/main.js

```diff
@@ -246,26 +221,55 @@ function searchPokeSeries(setName) {
 function searchPokeSet(setId) {
   const xhr = new XMLHttpRequest();
   displayLoading();
-  const targetUrl = encodeURIComponent('https://api.pokemontcg.io/v2/cards/?q=set.id:' + setId + '&pageSize=8&page=' + pokeCount);
+  let targetUrl;
+  if (legality === true) {
+    targetUrl = encodeURIComponent('https://api.pokemontcg.io/v2/cards/?q=set.id:' + setId + ' legalities.standard:legal&pageSize=8&page=' + pokeCount);
+  } else {
+    targetUrl = encodeURIComponent('https://api.pokemontcg.io/v2/cards/?q=set.id:' + setId + '&pageSize=8&page=' + pokeCount);
+  }
   xhr.open('GET', 'https://lfz-cors.herokuapp.com/?url=' + targetUrl);
   xhr.setRequestHeader('X-Api-Key', 'f81270c6-9d17-41a7-90ff-04e77b2b4273');
   xhr.responseType = 'json';
   xhr.addEventListener('load', function () {
     data.cardData = xhr.response;
     searchName = setId;
     const $found = document.querySelectorAll('.found-poke');
+    const cardData1 = data.cardData.data;
     if ($found.length === 0) {
-      for (let i = 0; i < data.cardData.data.length; i++) {
-        const image = renderPokeSearch(data.cardData.data[i].images.large);
-        image.setAttribute('data-cardId', data.cardData.data[i].id);
+      for (let i = 0; i < cardData1.length; i++) {
+        const image = renderPokeSearch(cardData1[i].images.large);
+        image.setAttribute('data-cardId', cardData1[i].id);
         $pokeSearch.appendChild(image);
         image.classList.add('set-search');
         hideLoading();
       }
+    } else if (cardData1.length < 8) {
+      for (let i = 0; i < cardData1.length; i++) {
+        $found[i].src = cardData1[i].images.large;
+        $found[i].setAttribute('data-cardId', cardData1[i].id);
+        $found[i].classList.add('set-search');
+      }
+      for (let i = cardData1.length; i < $found.length; i++) {
+        $found[i].remove();
+      }
+      hideLoading();
+    } else if ($found.length < cardData1.length) {
+      for (let i = 0; i < $found.length; i++) {
+        $found[i].src = cardData1[i].images.large;
+        $found[i].setAttribute('data-cardId', cardData1[i].id);
+        $found[i].classList.add('set-search');
+      }
+      for (let i = $found.length; i < cardData1.length; i++) {
+        const image = renderPokeSearch(cardData1[i].images.large);
+        image.setAttribute('data-cardId', cardData1[i].id);
+        $pokeSearch.appendChild(image);
+        image.classList.add('set-search');
+      }
```

Comment:
You seem to be calling most of this code multiple times. Are you able to break these up into their own functions?